docs/src/man/comparisons.md

```diff
@@ -53,61 +59,116 @@ df[findall(==('c'), df.id), :]
-|                           | `df[['x','y']].mean()`                                                       | `combine(df, [:x, :y] .=> mean)`                          |
-|                           | `df.filter(regex=("^x")).mean()`                                             | `combine(df, r"^x" .=> mean)`                             |
-|                           | `df[['x', 'y']].agg([max, min])`                                             | `combine(df, ([:x, :y] .=> [maximum minimum])...)`        |
-| Multivariate function     | `df.assign(x_y_cor = np.corrcoef(df.x, df.y)[0,1])`                          | `transform(df, [:x, :y] => cor)`                          |
-| Row-wise                  | `df.assign(x_y_min = df.apply(lambda v: min(v.x, v.y), axis=1))`             | `transform(df, [:x, :y] => ByRow(min))`                   |
-|                           | `df.assign(x_y_argmax = df.apply(lambda v: df.columns[v.argmax()], axis=1))` | `transform(df, AsTable([:x,:y]) => ByRow(argmax))`        |
-| DataFrame as input        | `df.groupby('grp').head(2)`                                                  | `combine(d -> first(d, 2), groupby(df, :grp))`            |
-| DataFrame as output       | `df[['x']].agg(lambda x: [min(x), max(x)])`                                  | `combine(:x => x -> (x = [minimum(x), maximum(x)],), df)` |
+This section includes more complex examples.
+
+| Operation                                       | pandas                                                                       | DataFrames.jl                                                                     |
+|-------------------------------------------------|------------------------------------------------------------------------------|-----------------------------------------------------------------------------------|
+| Complex Function                                | `df[['z']].agg(lambda v: np.mean(np.cos(v)))`                                | `combine(df, :z => v -> mean(cos, skipmissing(v)))`                               |
+| Aggregate multiple columns                      | `df.agg({'x': max, 'y': min})`                                               | `combine(df, :x => maximum, :y => minimum)`                                       |
+|                                                 | `df[['x','y']].mean()`                                                       | `combine(df, [:x, :y] .=> mean)`                                                  |
+|                                                 | `df.filter(regex=("^x")).mean()`                                             | `combine(v -> mapcols(mean, v), select(df, r"[xy]"))`                             |
+| Multiple columns x multiple functions functions | `df[['x', 'y']].agg([max, min])`                                             | `DataFrame([(agg = "$f", x = f(df.x), y = f(df.y)) for f in [maximum, minimum]])` |
```

Comment:
So one trick is to use broadcating with a matrix:
```suggestion
| Multiple columns × multiple functions functions | `df[['x', 'y']].agg([max, min])`                                             | `combine(df, vec([:x, :y] .=> [maximum minimum]))` |
```
Maybe we should provide something simpler, but what?

Reply:
Wouldn't it be nice if `combine` accepts a column name array and a function array as separate arguments and build its own cartesian product? I'm not sure if `combine` could take this new case as it's already very complex. Maybe we can choose a different verb.

Just an idea:
```
julia> function agg(df, cols, funcs) 
           result = DataFrame()
           result.function = string.(funcs)
           for c in cols
               result[!, c] = [f(df[!, c]) for f in funcs]
           end
           return result
       end
agg (generic function with 1 method)

julia> agg(df, [:x, :y], [maximum, minimum])
2×3 DataFrame
│ Row │ function │ x     │ y     │
│     │ String   │ Int64 │ Int64 │
├─────┼──────────┼───────┼───────┤
│ 1   │ maximum  │ 6     │ 9     │
│ 2   │ minimum  │ 1     │ 4     │
```

Reply:
can you please open a separate issue for this? Currently the pattern to use is `vec([:x, :y] .=> [maximum minimum])`. If you feel it is too verbose please comment in a new issue. Thank you!

Reply:
Removed this example and opened a new separate issue to discuss.